packages/headless/src/features/commerce/facets/facet-set/facet-set-reducers.ts

```diff
@@ -0,0 +1,180 @@
+import {DateRangeRequest} from '../../../facets/range-facets/date-facet-set/interfaces/request';
+import {NumericRangeRequest} from '../../../facets/range-facets/numeric-facet-set/interfaces/request';
+import {Parameters} from '../../parameters/parameters-actions';
+import {CommerceFacetSetState} from './facet-set-state';
+import {
+  CategoryFacetRequest,
+  CategoryFacetValueRequest,
+  DateFacetRequest,
+  NumericFacetRequest,
+  RegularFacetRequest,
+} from './interfaces/request';
+import {RegularFacetValue} from './interfaces/response';
+
+export function restoreFromParameters(
+  state: CommerceFacetSetState,
+  action: {payload: Parameters}
+) {
+  for (const facetId of Object.keys(state)) {
+    delete state[facetId];
+  }
+
+  if (action.payload.f) {
+    restoreRegularFacets(state, action.payload.f);
+  }
+  if (action.payload.nf) {
+    restoreRangeFacets(state, action.payload.nf, 'numericalRange');
+  }
+  if (action.payload.df) {
+    restoreRangeFacets(state, action.payload.df, 'dateRange');
+  }
+  if (action.payload.cf) {
+    restoreCategoryFacets(state, action.payload.cf);
+  }
+}
+
+function restoreRegularFacets(
+  state: CommerceFacetSetState,
+  parameterFacets: Record<string, string[]>
+) {
+  const entries = Object.entries(parameterFacets);
+  for (const [facetId, values] of entries) {
+    state[facetId] = {
+      request: {
+        facetId,
+        field: facetId,
+        // TODO: Simplify facet schema to only require necessary properties
```

Comment:
What's the context behind that simplification ? What would be the JIRA that describe the simplification ?

Adding TODO like this is fine if there is a concrete plan to address them very soon, otherwise these types of comments should not be left in there.